CMakeLists.txt

```diff
@@ -60,6 +60,15 @@ if (NOT HAVE_CXX_ATOMICS_WITHOUT_LIB
     link_libraries (atomic)
 endif ()
 
+if (WIN32)
+    if(CMAKE_SYSTEM_VERSION AND CMAKE_SYSTEM_VERSION VERSION_LESS_EQUAL "6.3")
+        set(WINVER 0x0603) # Minimum supported is Windows 8.1
+    else()
+        set(WINVER 0x0A00) # Windows 10/11 by default
+    endif()
+    add_definitions(-DWINVER=${WINVER} -D_WIN32_WINNT=${WINVER})
+endif()
+
```

Comment:
Do we really need this so explicitly?

The macro I used here: https://github.com/savoirfairelinux/opendht/pull/750/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR72-R89 does exactly this to avoid some problems on Windows installations where MSVC does not recognise the system version correctly. This leads to problems with `inet_pton` and other functions in case the real system version and `_WIN32_WINNT` point to different parts of code in `libgnutls` and WINAPI network stack.

Reply:
I saw what you did, and considered multiple strategies to define `_WIN32_WINNT`, including the stackoverflow solution.

The main rationale behind this strategy is that today, the only Windows API level still supported by Microsoft is essentially `0x0A00` (Windows 10/11). Microsoft still offers some commercial support for older versions, and some users are still on older Windows versions, so we would still support building for `0x0603` (Windows 8.1) when explicitly configured using `CMAKE_SYSTEM_VERSION`.

Is that reasonable? I would like to understand better the potential issues where MSVC does not recognize the system version correctly.